src/client/datascience/interactive-window/nativeInteractiveWindow.ts

```diff
@@ -500,29 +510,106 @@ export class NativeInteractiveWindow implements IInteractiveWindowLoadable {
         const language =
             workspace.textDocuments.find((document) => document.uri.toString() === this.owner?.toString())
                 ?.languageId ?? PYTHON_LANGUAGE;
-        const notebookCell = new NotebookCellData(
+        const notebookCellData = new NotebookCellData(
             isMarkdown ? NotebookCellKind.Markup : NotebookCellKind.Code,
             strippedCode,
             isMarkdown ? MARKDOWN_LANGUAGE : language
         );
-        notebookCell.metadata = { interactiveWindowCellMarker };
+        notebookCellData.metadata = { interactiveWindowCellMarker };
 
         edit.replaceNotebookCells(
             notebookDocument.uri,
             new NotebookRange(notebookDocument.cellCount, notebookDocument.cellCount),
             [
-                notebookCell // TODO generalize to arbitrary languages and cell types
+                notebookCellData // TODO generalize to arbitrary languages and cell types
             ]
         );
         await workspace.applyEdit(edit);
+        const notebookCell = notebookDocument.cellAt(notebookDocument.cellCount - 1);
+
+        let result = true;
 
         // Request execution
-        await this.commandManager.executeCommand('notebook.cell.execute', {
-            ranges: [{ start: notebookDocument.cellCount - 1, end: notebookDocument.cellCount }],
-            document: notebookDocument.uri,
-            autoReveal: true
-        });
-        return true;
+        if (!debugInfo) {
+            await this.kernel?.executeCell(notebookCell);
+        } else {
+            const notebook = this.kernel?.notebook;
+            if (!notebook) {
+                return false;
+            }
+
+            try {
+                const finishedAddingCode = createDeferred<void>();
+
+                // Before we try to execute code make sure that we have an initial directory set
```

Comment:
How come you don't need to do this for the non debug case?

Reply:
The non debug case just uses `notebook.cell.execute`. I found that this `Deferred` was necessary for debugging a cell to ensure the debug session doesn't prematurely terminate.